lib/src/migrators/module.dart

```diff
@@ -635,8 +636,10 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
   void _patchRemovedColorFunction(String name, Expression arg,
       {FileSpan existingArgName}) {
     var parameter = removedColorFunctions[name];
-    var needsParens =
-        parameter.endsWith('-') && arg is BinaryOperationExpression;
+    var needsParens = parameter.endsWith('-') &&
+        (arg is BinaryOperationExpression ||
+            (arg is VariableExpression &&
```

Comment:
This should handle function expressions as well. Also, add a comment explaining why parentheses are needed here (just copying the one below is fine).

Reply:
Done